src/api/Comments.js

```diff
@@ -229,17 +242,21 @@ class Comments extends OffsetBasedAPI {
         fileId: string,
         permissions: BoxItemPermission,
         successCallback: Function,
-        errorCallback: Function,
-        ...rest: any
+        errorCallback: (e: ElementsXhrError, code: string) => void,
+        offset: number = DEFAULT_FETCH_START,
+        limit: number = DEFAULT_FETCH_END,
+        fields: Array<string> = COMMENTS_FIELDS_TO_FETCH,
+        shouldFetchAll: boolean = true,
     ): void {
+        this.errorCode = ERROR_CODE_FETCH_COMMENTS;
         try {
             this.checkApiCallValidity(PERMISSION_CAN_COMMENT, permissions, fileId);
         } catch (e) {
-            errorCallback(e);
+            errorCallback(e, this.errorCode);
             return;
         }
 
-        this.offsetGet(fileId, successCallback, errorCallback, ...rest);
+        this.offsetGet(fileId, successCallback, errorCallback, offset, limit, fields, shouldFetchAll);
```

Comment:
Is this change related to the error codes? It seems like the defaults should live in the parent class rather than being duplicated in each child.

Reply:
No not really, I can remove some of the defaults and just make them optional